File: DSA/Ishita-Nigam/Day42/Day42_Q2.java
import java.util.Scanner;

public class Day42_Q2 {
    public int maxDistToClosest(int[] seats) {
        int n = seats.length;
        int empty = 0;
        int result = 0;
        int idx1 = -1, idx2 = -1;
        
        for (int i = 0; i < n; ++i) {
            if (seats[i] == 1) {
                empty = 0;
                if (idx1 == -1) idx1 = i;
                idx2 = i;
            } else {
                empty++;
                result = Math.max(result, (empty + 1) / 2);
            }
        }
        result = Math.max(result, Math.max(idx1, n - 1 - idx2));
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the number of seats: ");
        int n = scanner.nextInt();
        int[] seats = new int[n];

        System.out.println("Enter the seat arrangement (0 for empty, 1 for occupied):");
        for (int i = 0; i < n; i++) {
            seats[i] = scanner.nextInt();
        }
        scanner.close();

        Day42_Q2 solution = new Day42_Q2();
        int result = solution.maxDistToClosest(seats);
        System.out.println("The maximum distance to the closest person is: " + result);
    }
}

File: DSA/Aarohi-Sharma/Day42_q2.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int n = seats.size();
        int maxi = 1, equalDis = 1;
        for(int i=0; i<n; i++){
            if(seats[i] == 0){
                int left = i-1, leftCount = 0;
                while(left > 0){
                    if(seats[left] == 0){
                        left--;
                    }
                    else{
                        break;
                    }
                }
                leftCount = i-left;
                int right = i+1, rightCount = 0;
                while(right < n){
                    if(seats[right] == 0){
                        right++;
                    }
                    else{
                        break;
                    }
                }
                rightCount = right - i;
                if(right >= n){
                    rightCount = leftCount;
                }
                if(left < 0){
                    leftCount = rightCount;
                }
                if(leftCount == rightCount){
                    equalDis = rightCount;
                }
                if(leftCount < rightCount){
                    maxi = max(maxi, leftCount);
                }
                if(rightCount < leftCount){
                    maxi = max(maxi, rightCount);
                }
                maxi = max(equalDis, maxi);
            }
        }
        return maxi;
    }
};
